mailbox/store/src/test/java/org/apache/james/mailbox/store/PreDeletionHooksTest.java

```diff
@@ -67,25 +75,57 @@
     private PreDeletionHook hook1;
     private PreDeletionHook hook2;
     private PreDeletionHooks testee;
-    private MetricFactory metricFactory;
-    private TimeMetric timeMetric;
+    private RecordingMetricFactory metricFactory;
+
+    public static class RecordingTimeMetric implements TimeMetric {
+        private final String name;
+        private final Stopwatch stopwatch = Stopwatch.createStarted();
+        private final Consumer<Long> publishCallback;
+
+        RecordingTimeMetric(String name, Consumer<Long> publishCallback) {
+            this.name = name;
+            this.publishCallback = publishCallback;
+        }
+
+        @Override
+        public String name() {
+            return name;
+        }
+
+        @Override
+        public long stopAndPublish() {
+            long elapsed = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+            publishCallback.accept(elapsed);
+            return elapsed;
+        }
+    }
+
+    public static class RecordingMetricFactory implements MetricFactory {
+        private final Multimap<String, Long> executionTimesInMs = Multimaps.synchronizedSetMultimap(HashMultimap.create());
```

Comment:
synchronizedSetMultimap is not working as you expect, you need to synchronize it yourself